analyzers/src/SonarAnalyzer.VisualBasic/Rules/UnusedStringBuilder.cs

```diff
@@ -0,0 +1,74 @@
+
+    protected override IList<InvocationExpressionSyntax> GetInvocations(VariableDeclaratorSyntax declaration) =>
+        declaration.Parent.Parent.Parent.DescendantNodes().OfType<InvocationExpressionSyntax>().ToList();
+
+    protected override IList<ReturnStatementSyntax> GetReturnStatements(VariableDeclaratorSyntax declaration) =>
+        declaration.Parent.Parent.Parent.DescendantNodes().OfType<ReturnStatementSyntax>().ToList();
+
+    protected override IList<InterpolationSyntax> GetInterpolatedStrings(VariableDeclaratorSyntax declaration) =>
+        declaration.Parent.Parent.Parent.DescendantNodes().OfType<InterpolationSyntax>().ToList();
+
+    protected override bool IsStringBuilderContentRead(IList<InvocationExpressionSyntax> invocations, ISymbol variableSymbol, SemanticModel semanticModel) =>
+        invocations.Any(x => IsSameVariable(x.Expression, variableSymbol, semanticModel) && StringBuilderAccessMethods.Contains(x.GetName()));
+
+    protected override bool IsPassedToMethod(IList<InvocationExpressionSyntax> invocations, ISymbol variableSymbol, SemanticModel semanticModel) =>
+        invocations.Any(x => x.ArgumentList.Arguments.Any(y => IsSameVariable(y.GetExpression(), variableSymbol, semanticModel)));
+
+    protected override bool IsReturned(IList<ReturnStatementSyntax> returnStatements, ISymbol variableSymbol, SemanticModel semanticModel) =>
+        returnStatements.Any(x => IsSameVariable(x.Expression, variableSymbol, semanticModel));
+
+    protected override bool IsWithinInterpolatedString(IList<InterpolationSyntax> interpolations, ISymbol variableSymbol, SemanticModel semanticModel) =>
+        interpolations.Any(x => IsSameVariable(x.Expression, variableSymbol, semanticModel));
+
+    protected override bool IsPropertyReferenced(VariableDeclaratorSyntax declaration, IList<InvocationExpressionSyntax> invocations, ISymbol variableSymbol, SemanticModel semanticModel) =>
+        invocations.Any(x => IsSameVariable(x.Expression, variableSymbol, semanticModel) && semanticModel.GetOperation(x).Kind is OperationKindEx.PropertyReference);
+
+    private static bool IsSameVariable(ExpressionSyntax expression, ISymbol variableSymbol, SemanticModel semanticModel)
+    {
+        var identifiers = expression.DescendantNodesAndSelf().OfType<IdentifierNameSyntax>();
+        if (!identifiers.Any() && expression.Ancestors().OfType<ConditionalAccessExpressionSyntax>().Any())
+        {
+            identifiers = expression.Ancestors().OfType<ConditionalAccessExpressionSyntax>().First().DescendantNodesAndSelf().OfType<IdentifierNameSyntax>();
+        }
+        return identifiers.Any(x => IsSameVariable(x, variableSymbol, semanticModel));
+    }
+
+    private static bool IsSameVariable(IdentifierNameSyntax identifier, ISymbol variableSymbol, SemanticModel semanticModel) =>
+        variableSymbol.Equals(semanticModel.GetSymbolInfo(identifier).Symbol);
```

Comment:
Because variableSymbol is an `ILocalSymbol` you can do a name check before the (expensive) "GetSymbolInfo" call, but only by taking the different naming conventions into account.

Other symbol kinds are more difficult to compare because different names can point to the same thing (e.g. ITypeSymbol: `int` is equivalent to `System.Int32`).